yang/vyatta-protocols-frr-zebra-v1.yang

```diff
@@ -0,0 +1,227 @@
+module vyatta-protocols-frr-zebra-v1 {
+  namespace "urn:vyatta.com:mgmt:vyatta-protocols-frr-zebra:1";
+  prefix vyatta-protocols-frr-zebra-v1;
+
+  import vyatta-protocols-v1 {
+    prefix protocols;
+  }
+  import vyatta-types-v1 {
+    prefix types;
+  }
+  import configd-v1 {
+    prefix configd;
+  }
+  import vyatta-interfaces-v1 {
+    prefix if;
+  }
+  import vyatta-interfaces-dataplane-v1 {
+    prefix interfaces-dataplane;
+  }
+  import vyatta-interfaces-loopback-v1 {
+    prefix interfaces-loopback;
+  }
+  import vyatta-interfaces-switch-v1 {
+    prefix interfaces-switch;
+  }
+
+  organization "Niral Networks Private Limited";
+  contact
+    "Niral Networks
+    Postal: Sigma Soft Tech Park, Whitefield Main Road,
+    Bengaluru, India 560066
+    Web: www.niralnetworks.com";
+
+  description
+    "Copyright (c) 2019-2020, Niral Networks Intellectual Property.
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or
+    without modification, are permitted provided that the following
+    conditions are met:
+
+    1. Redistributions of source code must retain the above copyright
+    notice, this list of conditions and the following disclaimer.
+    2. Redistributions in binary form must reproduce the above
+    copyright notice, this list of conditions and the following
+    disclaimer in the documentation and/or other materials provided
+    with the distribution.
+    3. Neither the name of the copyright holder nor the names of its
+    contributors may be used to endorse or promote products derived
+    from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    'AS IS' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+
+    SPDX-License-Identifier: BSD-3-Clause
+    The YANG module package for vyatta-protocols-frr-zebra-v1";
+
+  revision 2021-03-02 {
+    description 
+      "Added link-params option.";
+  }
+  
+	typedef ieee-float {
```

Comment:
Both this and the float type below seem needlessly complicated. What's the intention here? Why was this method used rather than using a decimal64? Why are prefixing with 0x in some cases and not in others? The data model should abstract complexities in the underlying implementation, not expose them

Reply:
The prefix 0x was used so that we even can take hex decimal format as inputs. In FRR the input format takes both hex and floating-point and is of type IEEE-float. Should I stick to decimal64 instead of the above float types?
![image](https://user-images.githubusercontent.com/79829103/117123915-46983c80-adb5-11eb-93f1-4c527c02e802.png)

Reply:
Eg:
1.
![image](https://user-images.githubusercontent.com/79829103/117147909-54f55100-add3-11eb-8a42-9246492d57f6.png)
2.
![image](https://user-images.githubusercontent.com/79829103/117148334-c2a17d00-add3-11eb-8611-5255c5c6bdd0.png)
In Frr we were using both hex format and decimal format for taking input values like here in available-bandwidth

Reply:
The only difference between your ieee-float and float types is the 0x in front, both of them seem to accept hex digits. 
 What is the difference between those types supposed to be? Configuring a bandwidth in hexadecimal also seems very strange, do you know of anyone that actually does that?

Thinking about it more, I'm not even convinced that a decimal type should be required here. If there's a suitable unit that could be used in all cases, you could just have a uint with that unit. If not, possibly something like mutually exclusive containers named after the unit containing a leaf with the value would be a good way of modelling this. @wivory, @jsouthworth any thoughts?

Reply:
People do not generally use hexadecimal for setting up the values. Earlier I tried to keep the existing functionality as per FRR. However, I feel a simple uint32 would do and fit in all cases but I am open to hearing if you have any other suggestions @wivory, @jsouthworth.